src/main/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -188,7 +188,10 @@ public void addMetaDataLine(final VCFHeaderLine headerLine) {
      * @return all of the VCF header lines of the ##contig form in order, or an empty list if none were present
      */
     public List<VCFContigHeaderLine> getContigLines() {
-        return Collections.unmodifiableList(contigMetaData);
+        List<VCFContigHeaderLine> contigLines = new ArrayList<>(contigMetaData.size());
+        // this must preserve input order
+        contigLines.addAll(contigMetaData.values());
+        return Collections.unmodifiableList(contigLines);
```

Comment:
If you make this method return `Collection<>` you could avoid the copying and return `Collections.unmodifiableCollection(contigMetadata.values())`.

If not, you can avoid the two-step `new ...` `addAll()` by passing `contigMetaData.values()` to the `ArrayList` constructor.

In general, when writing APIs it's better to return `Collection` when possible so the API isn't tied to an implementation.